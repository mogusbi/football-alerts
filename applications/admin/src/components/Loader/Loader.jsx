import LinearProgress from '@material-ui/core/LinearProgress';
import PropTypes from 'prop-types';
import React, {memo} from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
`;

function Loader ({loading}) {
  return (
    <Wrapper>
      {loading && (
        <LinearProgress color='secondary' />
      )}
    </Wrapper>
  );
}

Loader.propTypes = {
  loading: PropTypes.bool.isRequired
};

export default memo(Loader);
