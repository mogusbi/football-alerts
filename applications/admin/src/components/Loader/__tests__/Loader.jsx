import {mount} from 'enzyme';
import React from 'react';
import Loader from '../Loader';

describe('Loader', () => {
  let wrapper;

  it('should render component when loading', () => {
    wrapper = mount(
      <Loader loading />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render component when loading', () => {
    wrapper = mount(
      <Loader loading={false} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
