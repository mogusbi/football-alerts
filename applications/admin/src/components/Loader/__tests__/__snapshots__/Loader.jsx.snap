// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should not render component when loading 1`] = `
.c0 {
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
}

<Loader
  loading={false}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      />
    </StyledComponent>
  </styled.div>
</Loader>
`;

exports[`Loader should render component when loading 1`] = `
.c0 {
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
}

<Loader
  loading={true}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <WithStyles(LinearProgress)
          color="secondary"
        >
          <LinearProgress
            classes={
              Object {
                "bar": "MuiLinearProgress-bar-11",
                "bar1Buffer": "MuiLinearProgress-bar1Buffer-16",
                "bar1Determinate": "MuiLinearProgress-bar1Determinate-15",
                "bar1Indeterminate": "MuiLinearProgress-bar1Indeterminate-14",
                "bar2Buffer": "MuiLinearProgress-bar2Buffer-18",
                "bar2Indeterminate": "MuiLinearProgress-bar2Indeterminate-17",
                "barColorPrimary": "MuiLinearProgress-barColorPrimary-12",
                "barColorSecondary": "MuiLinearProgress-barColorSecondary-13",
                "buffer": "MuiLinearProgress-buffer-6",
                "colorPrimary": "MuiLinearProgress-colorPrimary-2",
                "colorSecondary": "MuiLinearProgress-colorSecondary-3",
                "dashed": "MuiLinearProgress-dashed-8",
                "dashedColorPrimary": "MuiLinearProgress-dashedColorPrimary-9",
                "dashedColorSecondary": "MuiLinearProgress-dashedColorSecondary-10",
                "determinate": "MuiLinearProgress-determinate-4",
                "indeterminate": "MuiLinearProgress-indeterminate-5",
                "query": "MuiLinearProgress-query-7",
                "root": "MuiLinearProgress-root-1",
              }
            }
            color="secondary"
            variant="indeterminate"
          >
            <div
              className="MuiLinearProgress-root-1 MuiLinearProgress-colorSecondary-3 MuiLinearProgress-indeterminate-5"
              role="progressbar"
            >
              <div
                className="MuiLinearProgress-bar-11 MuiLinearProgress-barColorSecondary-13 MuiLinearProgress-bar1Indeterminate-14"
                style={Object {}}
              />
              <div
                className="MuiLinearProgress-bar-11 MuiLinearProgress-barColorSecondary-13 MuiLinearProgress-bar2Indeterminate-17"
                style={Object {}}
              />
            </div>
          </LinearProgress>
        </WithStyles(LinearProgress)>
      </div>
    </StyledComponent>
  </styled.div>
</Loader>
`;
